paper-api/src/main/java/org/bukkit/scoreboard/ScoreHolder.java

```diff
@@ -0,0 +1,25 @@
+package org.bukkit.scoreboard;
+
+import io.papermc.paper.InternalAPIBridge;
+import net.kyori.adventure.text.Component;
+import org.bukkit.OfflinePlayer;
+import org.jetbrains.annotations.ApiStatus;
+import org.jspecify.annotations.NullMarked;
+
+@ApiStatus.Experimental
+@NullMarked
+public interface ScoreHolder {
```

Comment:
Javadocs are missing on everything but also, this entire type is pretty annoying.

The fact that this *always* wraps means consumers cannot do *anything* with this beyond the exposed methods. A scoreholder is never e.g. instanceOf a Player or Entity.

I think this would be better off as a parent type to Entity and a separate implementation for those string-only implementations.